src/client.ts

```diff
@@ -1,91 +1,90 @@
-      )
+  private getSubsquidUrl = (options: SubsquidClientOptions): string => {
+    if ('network' in options) {
+      const networkUrl = NETWORK_CONFIG[options.network as keyof typeof NETWORK_CONFIG]
+      if (!networkUrl) {
+        throw new Error(
+          `Unsupported network: ${options.network}. Supported networks are: ${SUPPORTED_NETWORKS.join(', ')}`
+        )
+      }
+      return networkUrl
+    }
+    if ('customSubsquidUrl' in options) {
+      const { customSubsquidUrl } = options
+      if (!customSubsquidUrl || customSubsquidUrl.trim() === '') {
+        throw new Error('customSubsquidUrl cannot be empty')
+      }
+
+      try {
+        const url = new URL(customSubsquidUrl)
+        return url.toString()
+      } catch (error) {
+        throw new Error(`Invalid URL format: ${customSubsquidUrl}`)
+      }
     }
-    return configUrl
+    throw new Error(
+      'Invalid configuration. Provide either { network } or { customSubsquidUrl }.'
+    )
   }
 
   /**
    * Generic request method for GraphQL queries using fetch with type safety
    * @param query - The GraphQL query to execute
    * @returns Promise that resolves to the query result
    */
-  request = async <T>(query: string): Promise<T> => {
-    try {
-      const requestBody = JSON.stringify({ query })
+  async request<T>(query: string): Promise<T> {
+    const requestBody = JSON.stringify({ query })
+    const response = await fetch(this.clientUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: requestBody,
+    })
 
-      const response = await fetch(this.clientUrl, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: requestBody,
-      })
-
-      if (!response.ok) {
-        throw new Error(`Network response was not ok: ${response.statusText}`)
-      }
-
-      const result = await response.json()
-
-      if (result.errors) {
-        throw new Error(`GraphQL errors: ${JSON.stringify(result.errors)}`)
-      }
+    if (!response.ok) {
+      throw new Error(`Network request failed: ${response.status} ${response.statusText}`)
+    }
 
-      return result.data as T
-    } catch (error) {
-      console.error('Error in request', error)
-      throw error
-    };
+    const result = await response.json()
+    return result.data as T
   }
 
   /**
    * Generic query method that can handle any entity type with proper type safety
    * @param input - The query input to build and execute
    * @returns Promise that resolves to the query result
    */
-  async query (
-    input: string
-  ): Promise<unknown> {
-    try {
-      const queryStr = this.queryBuilder.build(input)
-      return this.request(queryStr)
-    } catch (error) {
-      console.error('Error in query', error)
-      throw error
-    }
+  async query<T extends QueryInput> (
+    input: T & Record<Exclude<keyof T, keyof EntityQueryMap>, never>
```

Comment:
Maybe extract `input` type into something more readable, something that represents the input type better